### PR TITLE
first enable gocritic and exclude all components that are not passing…

### LINKED ISSUE #9109: [all] Enable more linters on this repository
The opentelemetry-collector repository has two checks that are not enabled here. This is a meta issue to track these.

### Errcheck

- [x] #9750
- [x] #9749

### Gocritic

- [x] #10465
- [x] #10466

### Depguard (for sync/atomic):

See more context at open-telemetry/opentelemetry-collector#5215.

- [x] #9810
- [x] #9778

### Exhaustive

We also have an existing issue for checking coverage for all metric types that can be tackled with a linter:

- [ ] #7532
- [x] #23242
- [x] #23266

### Vulncheck

- [x] #19154

### Possible additional work

Some more linters that could be enabled here (these are not in the core repo): deadcode, structcheck, unused, ineffassign, tenv, go-errorlint.

## Approach

Fixing all errors at once is a lot of work, so I would suggest doing things in several steps:

1. A first PR enables the linter, excluding it on any components that are not passing right now.
2. Subsequent PRs make it pass on individual components and progressively remove the exclusion rules on faulty components. We can create individual issues for each linter and each component, depending on how hard they are to fix.

### LINKED ISSUE #10465: [all] Enable `gocritic` linter on golangci-lint

### PR DESCRIPTION
**Pull request by fatsheep9146**
**Description:** <Describe what has changed.>
Fix #10465 as @mx-psi suggested, first enable `gocritic` and exclude all components that are not passing right now.

**Link to tracking Issue:** <Issue number if applicable>
#10465

Signed-off-by: Ziqi Zhao <zhaoziqi9146@gmail.com>

### CODE REVIEW

File: receiver/awscontainerinsightreceiver/internal/ecsInfo/ecsinfo_test.go [hunk 1]
```diff
@@ -1,16 +1,16 @@
-//Copyright  OpenTelemetry Authors
```
Comment (mx-psi) [inline receiver/awscontainerinsightreceiver/internal/ecsInfo/ecsinfo_test.go:1]: These changes look unrelated

Comment (fatsheep9146) [reply receiver/awscontainerinsightreceiver/internal/ecsInfo/ecsinfo_test.go:1]: Yes. But if not change like this, `gocritic` will check those comment as not passed, and can't be escaped by nolint , so I directly modify those comments to correct format.

**Comment by mx-psi** [comment]
This should not fix the #9109 issue, but fix #10465 instead. Can you edit the description to reflect this?

File: receiver/awscontainerinsightreceiver/internal/ecsInfo/ecsinfo_test.go (hunk 1 above)
Comment (mx-psi) [reply receiver/awscontainerinsightreceiver/internal/ecsInfo/ecsinfo_test.go:1]: Thanks! That makes sense, we can keep it then

**Comment by fatsheep9146** [comment]
> This should not fix the #9109 issue, but fix #10465 instead. Can you edit the description to reflect this?

Done

**Review by mx-psi — APPROVED** [review]

**Comment by fatsheep9146** [comment]
@codeboten could you merge this?

File: receiver/awscontainerinsightreceiver/internal/ecsInfo/ecsinfo_test.go [hunk 2]
```diff
@@ -1,16 +1,16 @@
...
-//WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-//See the License for the specific language governing permissions and
-//limitations under the License.
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
 
 package ecsinfo
```
Comment (TylerHelmuth) [inline receiver/awscontainerinsightreceiver/internal/ecsInfo/ecsinfo_test.go:15]: Same with this package

File: receiver/awscontainerinsightreceiver/internal/ecsInfo/ecsinfo.go [hunk 3]
```diff
@@ -1,16 +1,16 @@
...
-//WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-//See the License for the specific language governing permissions and
-//limitations under the License.
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
 
 package ecsinfo // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awscontainerinsightreceiver/internal/ecsInfo"
```
[the remark this answers is not in the document]
Comment (fatsheep9146) [reply receiver/awscontainerinsightreceiver/internal/ecsInfo/ecsinfo.go:15]: I try to use `nolint` at first, but it failed to escape these comments. So I have to correct those comments directly without using nolint.

File: receiver/awscontainerinsightreceiver/internal/ecsInfo/ecsinfo_test.go (hunk 2 above)
Comment (fatsheep9146) [reply receiver/awscontainerinsightreceiver/internal/ecsInfo/ecsinfo_test.go:15]: Same reason as the ecsinfo.go

**Review by djaglowski — APPROVED** [review]
